examples/cpp/dds/TypeIntrospectionExample/arg_configuration.h

```diff
@@ -0,0 +1,258 @@
+    }
+
+};
+
+enum optionIndex
+{
+    UNKNOWN_OPT,
+    HELP,
+    TOPIC_NAME,
+    DOMAIN_ID,
+    DATA_TYPE,
+    DATA_TYPE_GENERATOR,
+    SAMPLES,
+    TYPE_OBJECT,
+    TYPE_INFORMATION,
+    TYPE_INFORMATION_DISABLE,
+};
+
+const option::Descriptor usage[] = {
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,
+      "Usage: TypeIntrospectionExample <publisher|subscriber>\n\nGeneral options:" },
+    { HELP,    0, "h", "help",               Arg::None,      "  -h \t--help  \tProduce help message." },
+
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nPublisher options:"},
+
+    { TOPIC_NAME, 0, "t", "topic",                  Arg::String,
+      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: TypeIntrospectionTopic)." },
+    { DATA_TYPE, 0, "x", "type",                  Arg::DataType,
+      "  -x <data_type_name> \t--type=<data_type_name>  \tTopic Data Type name (Default: hw). "
+      "hw            -> HelloWorld data type (one string and one integer). "
+      "array         -> Data type with an array (one uint and an array of size 3 integer). "
+      "seq           -> Data type with a sequence (one uint and a sequence of size 3 shorts). "
+      "struct        -> Complex data type with an internal struct (3 integers). "
+      "plain         -> HelloWorld struct (index and string) but string is an array of 20 chars, so data type is "
+      "Plain. "
+      "key           -> Struct with an index, a keyed string value and 1 integer. "
+      "complex_array -> Data type with an array of non-primitive objects. "
+      "complex_seq   -> Data type with a sequence of non-primitive objects.  "
+      "super_complex -> Complex data type composed of several of the other types at multiple levels.  "
+    },
+    { DATA_TYPE_GENERATOR, 0, "g", "generator",                  Arg::Generator,
+      "  -g <generator_name> \t--generator=<generator_name>  \tData Type Generator (Default: gen). "
+      "gen   -> Use IDL file and Fast DDS Gen to generate Data Type introspection information. "
+      "xml   -> Use XML file to generate Data Type introspection information. "
+      "code  -> Use DynamicTypes API to generate Data Type introspection information. "
+    },
+    { DOMAIN_ID, 0, "d", "domain",                Arg::Numeric,
+      "  -d <id> \t--domain=<id>  \tDDS domain ID (Default: 0)." },
+    { SAMPLES, 0, "s", "samples",              Arg::Numeric,
+      "  -s <num> \t--samples=<num>  \tNumber of samples to send (Default: 0 => infinite samples)." },
+    { TYPE_OBJECT, 0, "o", "type-object",              Arg::None,
+      "  -o \t--type-object=  \tUse Type Object to send Data Type Introspection info." },
+    { TYPE_INFORMATION, 1, "i", "type-information",              Arg::None,
+      "  -i \t--type-information=  \tUse Type Information to send Data Type Introspection info." },
+    { TYPE_INFORMATION_DISABLE, 0, "i", "type-information-disable",              Arg::None,
```

Comment:
Please check this works as expected: verify if enumeration should coincide and/or second argument should be 0 or 1. Also, perhaps one of them should not use `-i`.